diff -Naur tempered-9999.old/CMakeLists.txt tempered-9999/CMakeLists.txt
--- tempered-9999.old/CMakeLists.txt	2012-10-15 20:41:55.058113586 +0200
+++ tempered-9999/CMakeLists.txt	2012-10-15 20:51:33.401102699 +0200
@@ -38,7 +38,7 @@
 )
 
 if (BUILD_HIDAPI_SHARED)
-	find_library(HIDAPI_LIB NAMES hidapi-hidraw hidapi-libusb
+	find_library(HIDAPI_LIB NAMES hidapi-libusb hidapi-hidraw
 		PATHS ../hidapi ../hidapi.git
 		PATH_SUFFIXES linux libusb mac
 		DOC "The location of the HIDAPI shared library file"
