diff -Naur old/rygel-0.12.7/configure work/rygel-0.12.7/configure
--- old/rygel-0.12.7/configure	2012-01-18 18:28:09.000000000 +0100
+++ work/rygel-0.12.7/configure	2012-09-07 11:45:27.355358073 +0200
@@ -769,6 +769,8 @@
 GEE_CFLAGS
 LIBGIO_LIBS
 LIBGIO_CFLAGS
+GSTREAMER_INTERFACES_LIBS
+GSTREAMER_INTERFACES_CFLAGS
 GSTREAMER_BASE_LIBS
 GSTREAMER_BASE_CFLAGS
 LIBGSTREAMER_LIBS
@@ -948,6 +950,8 @@
 LIBGSTREAMER_LIBS
 GSTREAMER_BASE_CFLAGS
 GSTREAMER_BASE_LIBS
+GSTREAMER_INTERFACES_CFLAGS
+GSTREAMER_INTERFACES_LIBS
 LIBGIO_CFLAGS
 LIBGIO_LIBS
 GEE_CFLAGS
@@ -1664,6 +1668,10 @@
               C compiler flags for GSTREAMER_BASE, overriding pkg-config
   GSTREAMER_BASE_LIBS
               linker flags for GSTREAMER_BASE, overriding pkg-config
+  GSTREAMER_INTERFACES_CFLAGS
+              C compiler flags for GSTREAMER_INTERFACES, overriding pkg-config
+  GSTREAMER_INTERFACES_LIBS
+              linker flags for GSTREAMER_INTERFACES, overriding pkg-config
   LIBGIO_CFLAGS
               C compiler flags for LIBGIO, overriding pkg-config
   LIBGIO_LIBS linker flags for LIBGIO, overriding pkg-config
@@ -13070,6 +13078,97 @@
 fi
 
 pkg_failed=no
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for GSTREAMER_INTERFACES" >&5
+$as_echo_n "checking for GSTREAMER_INTERFACES... " >&6; }
+
+if test -n "$GSTREAMER_INTERFACES_CFLAGS"; then
+    pkg_cv_GSTREAMER_INTERFACES_CFLAGS="$GSTREAMER_INTERFACES_CFLAGS"
+ elif test -n "$PKG_CONFIG"; then
+    if test -n "$PKG_CONFIG" && \
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gstreamer-interfaces-0.10 >= \$GSTREAMER_REQUIRED\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED") 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+  pkg_cv_GSTREAMER_INTERFACES_CFLAGS=`$PKG_CONFIG --cflags "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
+else
+  pkg_failed=yes
+fi
+ else
+    pkg_failed=untried
+fi
+if test -n "$GSTREAMER_INTERFACES_LIBS"; then
+    pkg_cv_GSTREAMER_INTERFACES_LIBS="$GSTREAMER_INTERFACES_LIBS"
+ elif test -n "$PKG_CONFIG"; then
+    if test -n "$PKG_CONFIG" && \
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gstreamer-interfaces-0.10 >= \$GSTREAMER_REQUIRED\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED") 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+  pkg_cv_GSTREAMER_INTERFACES_LIBS=`$PKG_CONFIG --libs "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED" 2>/dev/null`
+		      test "x$?" != "x0" && pkg_failed=yes
+else
+  pkg_failed=yes
+fi
+ else
+    pkg_failed=untried
+fi
+
+
+
+if test $pkg_failed = yes; then
+   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+
+if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
+        _pkg_short_errors_supported=yes
+else
+        _pkg_short_errors_supported=no
+fi
+        if test $_pkg_short_errors_supported = yes; then
+	        GSTREAMER_INTERFACES_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED" 2>&1`
+        else
+	        GSTREAMER_INTERFACES_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED" 2>&1`
+        fi
+	# Put the nasty error message in config.log where it belongs
+	echo "$GSTREAMER_INTERFACES_PKG_ERRORS" >&5
+
+	as_fn_error $? "Package requirements (gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED) were not met:
+
+$GSTREAMER_INTERFACES_PKG_ERRORS
+
+Consider adjusting the PKG_CONFIG_PATH environment variable if you
+installed software in a non-standard prefix.
+
+Alternatively, you may set the environment variables GSTREAMER_INTERFACES_CFLAGS
+and GSTREAMER_INTERFACES_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details." "$LINENO" 5
+elif test $pkg_failed = untried; then
+     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
+$as_echo "no" >&6; }
+	{ { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
+$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
+as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
+is in your PATH or set the PKG_CONFIG environment variable to the full
+path to pkg-config.
+
+Alternatively, you may set the environment variables GSTREAMER_INTERFACES_CFLAGS
+and GSTREAMER_INTERFACES_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details.
+
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.
+See \`config.log' for more details" "$LINENO" 5; }
+else
+	GSTREAMER_INTERFACES_CFLAGS=$pkg_cv_GSTREAMER_INTERFACES_CFLAGS
+	GSTREAMER_INTERFACES_LIBS=$pkg_cv_GSTREAMER_INTERFACES_LIBS
+        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+
+fi
+
+pkg_failed=no
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for LIBGIO" >&5
 $as_echo_n "checking for LIBGIO... " >&6; }
 
diff -Naur old/rygel-0.12.7/configure.ac work/rygel-0.12.7/configure.ac
--- old/rygel-0.12.7/configure.ac	2012-01-18 18:19:53.000000000 +0100
+++ work/rygel-0.12.7/configure.ac	2012-09-07 11:31:38.858103815 +0200
@@ -62,6 +62,7 @@
                                  gupnp-dlna-1.0 <= $GUPNP_DLNA_MAX_VERSION)
 PKG_CHECK_MODULES(LIBGSTREAMER, gstreamer-0.10 >= $GSTREAMER_REQUIRED)
 PKG_CHECK_MODULES(GSTREAMER_BASE, gstreamer-base-0.10 >= $GSTREAMER_REQUIRED)
+PKG_CHECK_MODULES(GSTREAMER_INTERFACES, gstreamer-interfaces-0.10 >= $GSTREAMER_REQUIRED)
 PKG_CHECK_MODULES(LIBGIO, gio-2.0 >= $GIO_REQUIRED)
 PKG_CHECK_MODULES(GEE, gee-1.0 >= $GEE_REQUIRED)
 PKG_CHECK_MODULES(UUID, uuid >= $UUID_REQUIRED)
diff -Naur old/rygel-0.12.7/src/plugins/playbin/Makefile.in work/rygel-0.12.7/src/plugins/playbin/Makefile.in
--- old/rygel-0.12.7/src/plugins/playbin/Makefile.in	2012-01-18 18:28:13.000000000 +0100
+++ work/rygel-0.12.7/src/plugins/playbin/Makefile.in	2012-09-07 11:34:07.502072950 +0200
@@ -166,6 +166,8 @@
 GSTREAMER_APP_LIBS = @GSTREAMER_APP_LIBS@
 GSTREAMER_BASE_CFLAGS = @GSTREAMER_BASE_CFLAGS@
 GSTREAMER_BASE_LIBS = @GSTREAMER_BASE_LIBS@
+GSTREAMER_INTERFACES_CFLAGS = @GSTREAMER_INTERFACES_CFLAGS@
+GSTREAMER_INTERFACES_LIBS = @GSTREAMER_INTERFACES_LIBS@
 GSTREAMER_TAG_CFLAGS = @GSTREAMER_TAG_CFLAGS@
 GSTREAMER_TAG_LIBS = @GSTREAMER_TAG_LIBS@
 GTK_CFLAGS = @GTK_CFLAGS@
@@ -321,7 +323,7 @@
 rygeldir = $(top_srcdir)/src/rygel
 MAINTAINERCLEANFILES = Makefile.in
 AM_CFLAGS = $(LIBGUPNP_CFLAGS) $(LIBGUPNP_AV_CFLAGS) \
-	$(LIBGSTREAMER_CFLAGS) $(GSTREAMER_BASE_CFLAGS) $(GEE_CFLAGS) \
+	$(LIBGSTREAMER_CFLAGS) $(GSTREAMER_BASE_CFLAGS) $(GSTREAMER_INTERFACES_CFLAGS) $(GEE_CFLAGS) \
 	$(UUID_CFLAGS) -I$(rygeldir) -DDATA_DIR='"$(shareddir)"' \
 	-DSYS_CONFIG_DIR='"$(sysconfigdir)"' \
 	-DPLUGIN_DIR='"$(plugindir)"' -DDESKTOP_DIR='"$(desktopdir)"' \
@@ -331,6 +333,7 @@
 	$(LIBGUPNP_AV_LIBS) \
 	$(LIBGSTREAMER_LIBS) \
 	$(GSTREAMER_BASE_LIBS) \
+	$(GSTREAMER_INTERFACES_LIBS) \
 	$(GEE_LIBS) \
 	$(UUID_LIBS)
 
diff -Naur old/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.c work/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.c
--- old/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.c	2012-01-18 18:29:41.000000000 +0100
+++ work/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.c	2012-09-06 18:12:33.152479801 +0200
@@ -28,6 +28,9 @@
 #include <glib-object.h>
 #include <rygel.h>
 #include <gst/gst.h>
+#if GST_CHECK_VERSION(0,10,25)
+#include <gst/interfaces/streamvolume.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 
@@ -93,8 +96,6 @@
 static inline GstCaps* _dynamic_get_caps0 (GstElement* obj);
 static inline gchar* _dynamic_get_uri1 (GstElement* obj);
 static inline void _dynamic_set_uri2 (GstElement* obj, const gchar* value);
-static inline gdouble _dynamic_get_volume3 (GstElement* obj);
-static inline void _dynamic_set_volume4 (GstElement* obj, gdouble value);
 static void rygel_playbin_player_finalize (GObject* obj);
 static void _vala_rygel_playbin_player_get_property (GObject * object, guint property_id, GValue * value, GParamSpec * pspec);
 static void _vala_rygel_playbin_player_set_property (GObject * object, guint property_id, const GValue * value, GParamSpec * pspec);
@@ -489,33 +490,19 @@
 }
 
 
-static inline gdouble _dynamic_get_volume3 (GstElement* obj) {
-	gdouble result;
-	g_object_get (obj, "volume", &result, NULL);
-	return result;
-}
-
-
 static gdouble rygel_playbin_player_real_get_volume (RygelMediaPlayer* base) {
 	gdouble result;
 	RygelPlaybinPlayer* self;
 	GstElement* _tmp0_;
-	gdouble _tmp1_;
-	gdouble _tmp2_;
+	gdouble _tmp1_ = 0.0;
 	self = (RygelPlaybinPlayer*) base;
 	_tmp0_ = self->priv->playbin;
-	_tmp1_ = _dynamic_get_volume3 (_tmp0_);
-	_tmp2_ = _tmp1_;
-	result = _tmp2_;
+	_tmp1_ = gst_stream_volume_get_volume (GST_IS_STREAM_VOLUME (_tmp0_) ? ((GstStreamVolume*) _tmp0_) : NULL, GST_STREAM_VOLUME_FORMAT_CUBIC);
+	result = _tmp1_;
 	return result;
 }
 
 
-static inline void _dynamic_set_volume4 (GstElement* obj, gdouble value) {
-	g_object_set (obj, "volume", value, NULL);
-}
-
-
 static void rygel_playbin_player_real_set_volume (RygelMediaPlayer* base, gdouble value) {
 	RygelPlaybinPlayer* self;
 	GstElement* _tmp0_;
@@ -524,7 +511,7 @@
 	self = (RygelPlaybinPlayer*) base;
 	_tmp0_ = self->priv->playbin;
 	_tmp1_ = value;
-	_dynamic_set_volume4 (_tmp0_, _tmp1_);
+	gst_stream_volume_set_volume (GST_IS_STREAM_VOLUME (_tmp0_) ? ((GstStreamVolume*) _tmp0_) : NULL, GST_STREAM_VOLUME_FORMAT_CUBIC, _tmp1_);
 	_tmp2_ = value;
 	g_debug ("rygel-playbin-player.vala:108: volume set to %f.", _tmp2_);
 	g_object_notify ((GObject *) self, "volume");
@@ -587,11 +574,11 @@
 	G_OBJECT_CLASS (klass)->get_property = _vala_rygel_playbin_player_get_property;
 	G_OBJECT_CLASS (klass)->set_property = _vala_rygel_playbin_player_set_property;
 	G_OBJECT_CLASS (klass)->finalize = rygel_playbin_player_finalize;
-	g_object_class_override_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_PLAYBACK_STATE, "playback-state");
-	g_object_class_override_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_URI, "uri");
-	g_object_class_override_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_VOLUME, "volume");
-	g_object_class_override_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_DURATION, "duration");
-	g_object_class_override_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_POSITION, "position");
+	g_object_class_install_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_PLAYBACK_STATE, g_param_spec_string ("playback-state", "playback-state", "playback-state", NULL, G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB | G_PARAM_READABLE | G_PARAM_WRITABLE));
+	g_object_class_install_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_URI, g_param_spec_string ("uri", "uri", "uri", NULL, G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB | G_PARAM_READABLE | G_PARAM_WRITABLE));
+	g_object_class_install_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_VOLUME, g_param_spec_double ("volume", "volume", "volume", -G_MAXDOUBLE, G_MAXDOUBLE, 0.0, G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB | G_PARAM_READABLE | G_PARAM_WRITABLE));
+	g_object_class_install_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_DURATION, g_param_spec_int64 ("duration", "duration", "duration", G_MININT64, G_MAXINT64, 0, G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB | G_PARAM_READABLE));
+	g_object_class_install_property (G_OBJECT_CLASS (klass), RYGEL_PLAYBIN_PLAYER_POSITION, g_param_spec_int64 ("position", "position", "position", G_MININT64, G_MAXINT64, 0, G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB | G_PARAM_READABLE));
 }
 
 
diff -Naur old/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.vala work/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.vala
--- old/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.vala	2012-01-03 19:57:28.000000000 +0100
+++ work/rygel-0.12.7/src/plugins/playbin/rygel-playbin-player.vala	2012-09-06 18:09:28.165762705 +0200
@@ -100,11 +100,11 @@
 
     public double volume {
         get {
-            return this.playbin.volume;
+            return (this.playbin as StreamVolume).get_volume (StreamVolumeFormat.CUBIC);
         }
 
         set {
-            this.playbin.volume = value;
+            (this.playbin as StreamVolume).set_volume (StreamVolumeFormat.CUBIC, value);
             debug ("volume set to %f.", value);
         }
     }
