diff -Naur frandom-1.1.old/10-frandom.rules frandom-1.1/10-frandom.rules
--- frandom-1.1.old/10-frandom.rules	2011-01-29 16:29:22.000000000 +0100
+++ frandom-1.1/10-frandom.rules	2013-09-05 19:48:40.414768569 +0200
@@ -5,4 +5,4 @@
 # 50-udev.rules (0600) is too restrictive.
 #
 
-KERNEL=="[ef]random", NAME="%k", MODE="444", OPTIONS="last_rule"
+KERNEL=="[ef]random", MODE="444", OPTIONS="last_rule"
