--- etc/init.d/vdr.old	2011-06-06 14:38:15.053803589 +0200
+++ etc/init.d/vdr	2011-06-06 14:38:44.660530171 +0200
@@ -42,7 +42,7 @@
 	einfo "Preparing start of vdr:"
 	clear_logfile
 	init_params
-	add_daemonctrl_param --start --chdir ~vdr --exec ${VDR_BIN}
+	add_daemonctrl_param --start --chdir ~vdr --umask 002 --exec ${VDR_BIN}
 	init_plugin_loader start
 	
 	load_addons_prefixed pre-start || return 1
