diff -Naur vala-0.18.1.old/vapi/gio-2.0.vapi vala-0.18.1/vapi/gio-2.0.vapi
--- vala-0.18.1.old/vapi/gio-2.0.vapi	2012-09-16 19:17:36.000000000 +0200
+++ vala-0.18.1/vapi/gio-2.0.vapi	2013-06-28 20:26:06.086443882 +0200
@@ -500,7 +500,8 @@
 		[CCode (cname = "g_dbus_connection_new_for_address", finish_function = "g_dbus_connection_new_for_address_finish")]
 		public static async GLib.DBusConnection new_for_address (string address, GLib.DBusConnectionFlags flags, GLib.DBusAuthObserver? observer = null, GLib.Cancellable? cancellable = null) throws GLib.Error;
 		public uint register_object<T> (string object_path, T object) throws GLib.IOError;
-		public uint register_subtree (string object_path, GLib.DBusSubtreeVTable vtable, GLib.DBusSubtreeFlags flags, void* user_data, GLib.DestroyNotify user_data_free_func) throws GLib.Error;
+        [CCode (simple_generics = true)]
+		public uint register_subtree<G> (string object_path, GLib.DBusSubtreeVTable vtable, GLib.DBusSubtreeFlags flags, owned G data) throws GLib.Error;
 		public void remove_filter (uint filter_id);
 		public bool send_message (GLib.DBusMessage message, GLib.DBusSendMessageFlags flags, out uint32 out_serial) throws GLib.Error;
 		public async GLib.DBusMessage send_message_with_reply (GLib.DBusMessage message, GLib.DBusSendMessageFlags flags, int timeout_msec, uint32* out_serial = null, GLib.Cancellable? cancellable = null) throws GLib.IOError;
@@ -3344,24 +3345,24 @@
 	public delegate void BusNameVanishedCallback (GLib.DBusConnection connection, string name);
 	[CCode (cheader_filename = "gio/gio.h", instance_pos = 1.9)]
 	public delegate bool CancellableSourceFunc (GLib.Cancellable? cancellable = null);
-	[CCode (cheader_filename = "gio/gio.h", instance_pos = 6.9)]
-	public delegate GLib.Variant DBusInterfaceGetPropertyFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string property_name) throws GLib.Error;
-	[CCode (cheader_filename = "gio/gio.h", instance_pos = 7.9)]
-	public delegate void DBusInterfaceMethodCallFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string method_name, GLib.Variant parameters, GLib.DBusMethodInvocation invocation);
-	[CCode (cheader_filename = "gio/gio.h", instance_pos = 7.9)]
-	public delegate bool DBusInterfaceSetPropertyFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string property_name, GLib.Variant value) throws GLib.Error;
+	[CCode (cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate GLib.Variant DBusInterfaceGetPropertyFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string property_name, GLib.Error* error, void* user_data);
+	[CCode (cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate void DBusInterfaceMethodCallFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string method_name, GLib.Variant parameters, GLib.DBusMethodInvocation invocation, void* user_data);
+	[CCode (cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate bool DBusInterfaceSetPropertyFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string property_name, GLib.Variant value, GLib.Error* error, void* user_data);
 	[CCode (cheader_filename = "gio/gio.h", instance_pos = 3.9)]
 	public delegate GLib.DBusMessage DBusMessageFilterFunction (GLib.DBusConnection connection, owned GLib.DBusMessage message, bool incoming);
 	[CCode (cheader_filename = "gio/gio.h", instance_pos = 3.9)]
 	public delegate GLib.Type DBusProxyTypeFunc (GLib.DBusObjectManagerClient manager, string object_path, string? interface_name);
 	[CCode (cheader_filename = "gio/gio.h", instance_pos = 6.9)]
 	public delegate void DBusSignalCallback (GLib.DBusConnection connection, string sender_name, string object_path, string interface_name, string signal_name, GLib.Variant parameters);
-	[CCode (cheader_filename = "gio/gio.h", instance_pos = 6.9)]
-	public delegate unowned GLib.DBusInterfaceVTable? DBusSubtreeDispatchFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string node, void* out_user_data);
-	[CCode (array_length = false, array_null_terminated = true, cheader_filename = "gio/gio.h", instance_pos = 3.9)]
-	public delegate string[] DBusSubtreeEnumerateFunc (GLib.DBusConnection connection, string sender, string object_path);
-	[CCode (cheader_filename = "gio/gio.h", instance_pos = 4.9)]
-	public delegate GLib.DBusInterfaceInfo DBusSubtreeIntrospectFunc (GLib.DBusConnection connection, string sender, string object_path, string node);
+	[CCode (cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate unowned GLib.DBusInterfaceVTable? DBusSubtreeDispatchFunc (GLib.DBusConnection connection, string sender, string object_path, string interface_name, string node, void** out_user_data, void* user_data);
+	[CCode (array_length = false, array_null_terminated = true, cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate string[] DBusSubtreeEnumerateFunc (GLib.DBusConnection connection, string sender, string object_path, void* user_data);
+	[CCode (array_length = false, array_null_terminated = true, cheader_filename = "gio/gio.h", has_target = false)]
+	public delegate GLib.DBusInterfaceInfo[] DBusSubtreeIntrospectFunc (GLib.DBusConnection connection, string sender, string object_path, string node, void* user_data);
 	[CCode (cheader_filename = "gio/gio.h", instance_pos = 2.9)]
 	public delegate void FileProgressCallback (int64 current_num_bytes, int64 total_num_bytes);
 	[CCode (cheader_filename = "gio/gio.h", has_target = false)]
