--- lib/log.lib.php.old	2012-03-08 12:15:44.414700115 +0100
+++ lib/log.lib.php	2012-03-08 12:16:14.730463387 +0100
@@ -34,14 +34,13 @@
 	/* Set it up here to make sure it's _always_ the same */
 	$time		= time();
 	// Turn time into strings
-	$log_day	= date('Ymd', $time);
 	$log_time	= date('Y-m-d H:i:s', $time);
 
 	/* must have some name */
 	$log_name	= $log_name ? $log_name : 'ampache';
 	$username	= $username ? $username : 'ampache';
 
-	$log_filename	= Config::get('log_path') . "/$log_name.$log_day.log";
+	$log_filename	= Config::get('log_path') . "/$log_name.log";
 	$log_line	= "$log_time [$username] ($event_name) -> $event_description \n";
 
 	// Do the deed
