diff -Naur xine-0.7.10.orig/Makefile xine-0.7.10/Makefile
--- xine-0.7.10.orig/Makefile	2007-01-03 19:27:55.672892461 +0100
+++ xine-0.7.10/Makefile	2007-01-03 19:31:23.820754133 +0100
@@ -38,7 +38,7 @@
 INCLUDES += `xine-config --cflags` 
 
 # where to create fifos (xine expects them at /tmp/vdr-xine)
-VDR_XINE_FIFO_DIR ?= /tmp/vdr-xine
+VDR_XINE_FIFO_DIR ?= /var/vdr/xine/
 
 # enable to fully support yaepg plugin
 #VDR_XINE_SET_VIDEO_WINDOW = 1
@@ -51,7 +51,7 @@
 
 ### The version number of VDR's plugin API (taken from VDR's "config.h"):
 
-APIVERSION ?= $(shell (sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h ; sed -ne '/define VDRVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h) | sed -ne 1p)
+APIVERSION ?= $(shell (sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h ; sed -ne '/define VDRVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/vdr/config.h) | sed -ne 1p)
 
 ### The name of the distribution archive:
 
diff -Naur xine-0.7.10.orig/xineLib.c xine-0.7.10/xineLib.c
--- xine-0.7.10.orig/xineLib.c	2007-01-03 19:27:55.684893146 +0100
+++ xine-0.7.10/xineLib.c	2007-01-03 19:42:35.759045712 +0100
@@ -2052,7 +2052,7 @@
     ::memset(m_osdWindowSupportTransparency, 0, sizeof (m_osdWindowSupportTransparency));
 #endif    
 
-    string noSignalFileName = plugin->ConfigDirectory(PLUGIN_NAME_I18N);
+    string noSignalFileName = plugin->ConfigDirectory("../../../usr/share/vdr/xine");
 //    noSignalFileName += "/noSignal.pes";
     noSignalFileName += "/noSignal.mpg";
         
