diff -Naur ampache-9999.old/lib/log.lib.php ampache-9999/lib/log.lib.php
--- ampache-9999.old/lib/log.lib.php	2013-02-17 20:36:44.025145260 +0100
+++ ampache-9999/lib/log.lib.php	2013-02-17 20:39:49.625949419 +0100
@@ -28,14 +28,13 @@
     /* Set it up here to make sure it's _always_ the same */
     $time        = time();
     // Turn time into strings
-    $log_day    = date('Ymd', $time);
     $log_time    = date('Y-m-d H:i:s', $time);
 
     /* must have some name */
     $log_name    = $log_name ? $log_name : 'ampache';
     $username    = $username ? $username : 'ampache';
 
-    $log_filename    = Config::get('log_path') . "/$log_name.$log_day.log";
+    $log_filename    = Config::get('log_path') . "/$log_name.log";
     $log_line    = "$log_time [$username] ($event_name) -> $event_description \n";
 
     // Do the deed
