diff -Naur work.old/mod_bw.c work/mod_bw.c
--- work.old/mod_bw.c	2010-07-20 18:14:30.000000000 +0200
+++ work/mod_bw.c	2013-09-03 19:26:16.153691034 +0200
@@ -97,6 +97,7 @@
 #endif
 
 /* Compatibility for APR < 1 */
+/*
 #if ( defined(APR_MAJOR_VERSION) && (APR_MAJOR_VERSION < 1) )
     #define apr_atomic_inc32 apr_atomic_inc
     #define apr_atomic_dec32 apr_atomic_dec
@@ -104,6 +105,7 @@
     #define apr_atomic_cas32 apr_atomic_cas
     #define apr_atomic_set32 apr_atomic_set
 #endif
+*/
 
 /* Enum types of "from address" */
 enum from_type {
@@ -564,7 +566,7 @@
             return e[i].rate;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->connection->client_addr)) {
                 return e[i].rate;
             }
             break;
@@ -655,7 +657,7 @@
             return e[i].max;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->connection->client_addr)) {
                 return e[i].max;
             }
             break;
@@ -706,7 +708,7 @@
             return e[i].sid;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->connection->client_addr)) {
                 return e[i].sid;
             }
             break;
