diff -Naur GDriveFS-0.14.3.old/setup.py GDriveFS-0.14.3/setup.py
--- GDriveFS-0.14.3.old/setup.py	2015-03-28 18:32:30.000000000 +0100
+++ GDriveFS-0.14.3/setup.py	2015-09-15 14:57:41.212503085 +0200
@@ -36,7 +36,7 @@
     author_email='myselfasunder@gmail.com',
     url='https://github.com/dsoprea/GDriveFS',
     license='GPL 2',
-    packages=setuptools.find_packages(exclude=['dev', 'tests']),
+    packages=setuptools.find_packages(exclude=['dev', 'tests', 'tests.*']),
     include_package_data=True,
     package_data={
         'gdrivefs': [
