diff -Naur gdrivefs-9999.old/setup.py gdrivefs-9999/setup.py
--- gdrivefs-9999.old/setup.py	2013-12-02 18:43:04.772066444 +0100
+++ gdrivefs-9999/setup.py	2013-12-02 18:44:57.622994778 +0100
@@ -42,7 +42,7 @@
       author_email='myselfasunder@gmail.com',
       url='https://github.com/dsoprea/GDriveFS',
       license='GPL',
-      packages=find_packages(),
+      packages=find_packages(exclude=['tests', 'tests.*']),
       include_package_data=True,
       zip_safe=True,
       install_requires=[
@@ -59,8 +59,8 @@
       entry_points="""
       # -*- Entry points: -*-
       """,
-      cmdclass={'install': custom_install
-               },
+#      cmdclass={'install': custom_install
+#               },
       scripts=['gdrivefs/tools/gdfs',
                'gdrivefs/tools/gdfstool',
                'gdrivefs/tools/gdfsuninstall'],
