--- daemons/hw_atilibusb.c.old	2011-06-06 17:35:19.390171477 +0200
+++ daemons/hw_atilibusb.c	2011-06-06 17:36:41.140585030 +0200
@@ -104,6 +104,7 @@
 {
 	struct usb_device *usb_dev;
 	int pipe_fd[2] = { -1, -1 };
+	static char path[] = "/tmp";
 
 	LOGPRINTF(1, "initializing USB receiver");
 
@@ -153,6 +154,7 @@
 		logperror(LOG_ERR, "couldn't fork child process");
 		goto fail;
 	} else if (child == 0) {
+	        chdir(path);
 		usb_read_loop(pipe_fd[1]);
 	}
 
