diff -Naur ampache.old//AmpacheBrowser.py ampache/AmpacheBrowser.py
--- ampache.old//AmpacheBrowser.py	2011-11-19 16:28:12.175530097 +0100
+++ ampache/AmpacheBrowser.py	2011-11-19 16:28:30.202708286 +0100
@@ -83,7 +83,7 @@
 
 		width, height      = gtk.icon_size_lookup(gtk.ICON_SIZE_LARGE_TOOLBAR)
 		icon               = gtk.gdk.pixbuf_new_from_file_at_size(self.config.get("icon_filename"), width, height)
-		self.set_property( "icon",  icon) 
+		#self.set_property( "icon",  icon) 
 
 		shell              = self.get_property("shell")
 		self.db            = shell.get_property("db")
diff -Naur ampache.old//__init__.py ampache/__init__.py
--- ampache.old//__init__.py	2011-11-19 16:28:18.515943754 +0100
+++ ampache/__init__.py	2011-11-19 16:28:25.190380093 +0100
@@ -38,8 +38,14 @@
 		self.entry_type.sync_metadata = None
 		self.entry_type.category = rhythmdb.ENTRY_STREAM
 
-		group = rb.rb_source_group_get_by_name(self.config.get("group"))
-		if not group:
+		'''group = rb.rb_source_group_get_by_name(self.config.get("group"))'''
+		theme = gtk.icon_theme_get_default()
+		rb.append_plugin_source_path(theme, "/icons/")
+		width, height = gtk.icon_size_lookup(gtk.ICON_SIZE_LARGE_TOOLBAR)
+		#icon = rb.try_load_icon(theme, "ampache", width, 0)
+		group = rb.rb_display_page_group_get_by_id ("stores")
+
+		'''if not group:
 			group = rb.rb_source_group_register (
 				"ampache",
 				self.config.get("group"),
@@ -50,17 +56,30 @@
 		self.source = gobject.new (
 			AmpacheBrowser,
  			entry_type=self.entry_type,
+			plugin=self,
 			source_group=group,
  			name=self.config.get("name"),
  			shell=shell,
-		)
+		)'''
 
 		self.config.set("icon_filename", self.find_file(self.config.get("icon")))
+
+		#icon = rb.try_load_icon(theme, "ampache", width, 0)
+		icon = gtk.gdk.pixbuf_new_from_file_at_size(self.config.get("icon_filename"), width, height)
+
+		self.source = gobject.new (AmpacheBrowser,
+					   shell=shell,
+					   entry_type=self.entry_type,
+					   plugin=self,
+					   name=self.config.get("name"),
+					   pixbuf=icon)
+
 		self.source.activate(self.config)
 
 		shell.register_entry_type_for_source(self.source, self.entry_type)
-		shell.append_source(self.source, None)
-
+		#shell.append_source(self.source, None)
+		shell.append_display_page(self.source, group)
+		
 		ui_manager = shell.get_ui_manager()
 		action = gtk.Action('RefetchAmpache', 
 				    _('_Re-fetch Ampache Library'), 
@@ -84,9 +103,9 @@
                 del self.source
                 self.source = None
 
-		# clean up the UI
+		'''# clean up the UI
 		ui_manager = shell.get_ui_manager()
-		ui_manager.remove_ui(self.uid)
+		ui_manager.remove_ui(self.uid)'''
 
 	def create_configure_dialog(self):
 		glade_file = self.find_file("ampache-prefs.glade")
