--- ffnetdev-9999/osdworker.c.old	2010-05-28 16:16:35.293584481 +0200
+++ ffnetdev-9999/osdworker.c	2010-05-28 16:18:23.707584420 +0200
@@ -752,23 +752,28 @@
 						pxfmt.pad2 = 0;
 						
 						rfbServerInitMsg simsg;
+dsyslog("[ffnetdev] DEBUG1"); 
 						simsg.framebufferWidth = Swap16IfLE(720);
 						simsg.framebufferHeight = Swap16IfLE(576);
 						simsg.format = pxfmt;
 						simsg.nameLength = Swap32IfLE(7);
+dsyslog("[ffnetdev] DEBUG2"); 
 						if (!m_OSDClient->SafeWrite(&simsg, sizeof(simsg)))
 						{   
 						    state = NO_CLIENT;
 						    break;
 						}
 						
+dsyslog("[ffnetdev] DEBUG3"); 
 					 	CARD8 name[7];
 					 	strcpy((char*)&name, "VDR-OSD");
+dsyslog("[ffnetdev] DEBUG4"); 
 					 	if (!m_OSDClient->SafeWrite(&name, 7))
 						{   
 						    state = NO_CLIENT;
 						    break;
 						}
+dsyslog("[ffnetdev] DEBUG5"); 
 					 	
 						state = HANDSHAKE_OK;
 						break;
