--- src/video_out/video_out_opengl.c.old	2012-02-01 21:47:22.570217900 +0100
+++ src/video_out/video_out_opengl.c	2012-02-01 23:44:02.012692672 +0100
@@ -1102,6 +1102,7 @@
 	glFlush ();
 	CHECKERR ("post-clean");
 	XUnlockDisplay (this->display);
+        xprintf (this->xine, XINE_VERBOSITY_NONE, "***CLEANED\n");
 	if (! ret) {
           xprintf (this->xine, XINE_VERBOSITY_NONE,
 	           "video_out_opengl: rendering '%s' failed, switching to fallback\n",
