--- src/audio_out/audio_pulse_out.c.old	2010-09-08 12:07:59.397472521 +0200
+++ src/audio_out/audio_pulse_out.c	2010-09-08 12:18:36.782762367 +0200
@@ -235,7 +235,7 @@
   xine_stream_t            *stream;
   xine_list_iterator_t      ite;
 
-  data.right        = data.left = (int) (pa_sw_volume_to_linear(this->swvolume)*100);
+  data.right        = data.left = (int) ((double) ((this->swvolume > PA_VOLUME_NORM ? PA_VOLUME_NORM : this->swvolume) - PA_VOLUME_MUTED) / (PA_VOLUME_NORM - PA_VOLUME_MUTED) * 100 + 0.5);
 
   data.mute         = this->muted;
 
@@ -741,7 +741,7 @@
 
     case AO_PROP_PCM_VOL:
     case AO_PROP_MIXER_VOL:
-      result = (int) (pa_sw_volume_to_linear(this->swvolume)*100);
+      result = (int) ((double) ((this->swvolume > PA_VOLUME_NORM ? PA_VOLUME_NORM : this->swvolume) - PA_VOLUME_MUTED) / (PA_VOLUME_NORM - PA_VOLUME_MUTED) * 100 + 0.5);
       break;
   }
 
@@ -769,7 +769,7 @@
     case AO_PROP_PCM_VOL:
     case AO_PROP_MIXER_VOL:
 
-      this->swvolume = pa_sw_volume_from_linear((double)value/100.0);
+      this->swvolume = (int) ((double) value / 100 * (PA_VOLUME_NORM - PA_VOLUME_MUTED) + PA_VOLUME_MUTED + 0.5);
       pa_cvolume_set(&this->cvolume, pa_stream_get_sample_spec(this->stream)->channels, this->swvolume);
 
       o = pa_context_set_sink_input_volume(this->context, pa_stream_get_index(this->stream),
