--- src/audio_out/audio_pulse_out.c.old	2010-09-19 11:48:11.201222791 +0200
+++ src/audio_out/audio_pulse_out.c	2010-09-19 11:50:43.809674024 +0200
@@ -361,6 +361,7 @@
   pa_sample_spec ss;
   pa_channel_map cm;
   int r;
+  pa_buffer_attr attr;
 
   xprintf (this->xine, XINE_VERBOSITY_DEBUG,
            "audio_pulse_out: ao_open bits=%d rate=%d, mode=%d\n", bits, rate, mode);
@@ -486,6 +487,17 @@
     pa_threaded_mainloop_wait(this->mainloop);
   }
 
+  attr.maxlength = 8192 * this->bytes_per_frame; // old: 4194304
+  attr.tlength   = 4096 * this->bytes_per_frame; // old: 48000;
+  attr.prebuf    = 4096 * this->bytes_per_frame; // old: 44164;
+  attr.minreq    = 1024 * this->bytes_per_frame; // old: 3840;
+  attr.fragsize  = -1;
+
+  if (pa_stream_set_buffer_attr(this->stream, &attr, __xine_pa_stream_success_callback, this) == NULL) {
+    xprintf(this->xine, XINE_VERBOSITY_DEBUG, "audio_pulse_out: failed to set buffer attributes: %s\n", pa_strerror(pa_context_errno (this->context)));
+    goto fail;
+  }
+
   pa_threaded_mainloop_unlock(this->mainloop);
 
   /* Now we must handle a problem: at init time, xine might have tried to set the default volume value
