--- configure.ac	2009-01-15 15:45:01.000000000 +0200
+++ configure.ac.good	2009-01-15 15:11:39.000000000 +0200
@@ -1133,8 +1133,7 @@
 CC_CHECK_CFLAGS_APPEND([-Wstrict-aliasing=2 -Wstrict-aliasing], [break;])
 
 dnl This *has* to stay at the end as it can break some autoconf tests.
-CC_CHECK_CFLAGS_APPEND([-Werror=implicit-function-declaration dnl
-                        -Werror-implicit-function-declaration], [break;])
+CC_CHECK_CFLAGS_APPEND([-Werrorimplicit-function-declaration], [break;])
 
 CC_NOUNDEFINED
 
